File: AndroidLab/ch10/src/main/java/com/example/ch10/Test2Activity.java
package com.example.ch10;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.ch10.databinding.ActivityTest2Binding;

public class Test2Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);

        ActivityTest2Binding binding = ActivityTest2Binding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        binding.insertBtn.setOnClickListener(view -> {
            //유저 입력 데이터 획득..
            String name = binding.nameInputView.getText().toString();
            String address = binding.addressInputView.getText().toString();

            //테이블에 저장.. 테이블이 만들어졌는지는 신경쓸 필요가 없다..
            SQLiteDatabase db = new DBHelper(this).getWritableDatabase();
            //컬럼 데이터를 ContentValues 로 표현해서.. Map 이라고 생각.. key 를 column 명으로만..
            ContentValues values = new ContentValues();
            values.put("name", name);
            values.put("address", address);
            //저장..
            db.insert("tb_user", null, values);
            db.close();
        });

        binding.queryBtn.setOnClickListener(view -> {
            SQLiteDatabase db = new DBHelper(this).getReadableDatabase();
            //colums 가 null - 모든 컬럼 데이터..
            //select * from tb_user
            //db.rawQuery("select * from tb_user", null)
            Cursor cursor = db.query("tb_user", null, null, null,
                    null, null, null);
            String result = "";
            //한껀도 없다.. - while 한번도 실행안되고..
            //10 - while 문 순차적으로 10번..
            while(cursor.moveToNext()){
                result += cursor.getString(1);
                result += ":";
                result += cursor.getString(2);
                result += "\n";
            }
            binding.resultView.setText(result);
        });
    }
}

File: Android3/ch1/src/main/java/com/example/ch1/section9/Test2Activity.kt
package com.example.ch1.section9

import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.lifecycle.lifecycleScope
import com.example.ch1.R
import com.example.ch1.databinding.ActivityTest2Binding
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class Test2Activity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val binding = ActivityTest2Binding.inflate(layoutInflater)
        setContentView(binding.root)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.button3.setOnClickListener {
            finish()
        }
        binding.button2.setOnClickListener {
            //코루틴을 실행시키려면 먼저 scope 가 선언되어 있어야 한다..
            //- activity, fragment에서 CoroutineScope 를 구현해서.. 자체가 코루틴 스코프가 되거나..
            //- CoroutineScope() 함수를 이용해 스코프를 만들어 사용하거나..
            //액티비티, 프레그먼트 생존기간에만 동작해야 하는 코루프를 만들려면 액티비티 라이프사이클에서 신경..

            //이 부분을 도와 주기 위해서 안드로이드에서 추가된 scope 를 이용해서 개발..
            //액티비티, 프레그먼트, viewmodel 등과 동일 라이프사이클로 동작하는 scope...
            lifecycleScope.launch {
                repeat(5){
                    Log.d("kkang", "coroutine $it")
                    delay(1000)
                }
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("kkang", "activity onDestory")
    }
}